### PR TITLE
Adding ApplicationGateway.start(), .stop() and async versions

### PR DESCRIPTION
**Pull request by martinsawicki**

### CODE REVIEW

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java [hunk 1]
```diff
@@ -1235,4 +1237,28 @@ public ApplicationGatewayTier tier() {
...
+
+    @Override
+    public void stop() {
+        this.stopAsync().await();
+        this.refresh();
+    }
+
+    @Override
+    public Completable startAsync() {
+        return this.manager().inner().applicationGateways().startAsync(this.resourceGroupName(), this.name()).toCompletable();
```
Comment (anuchandy) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java]: you can call `refreshAsync()` from `startAsync()` and remove `refresh()` from `start()`

```java
@Override
public Completable startAsync() {
        this.manager().inner().applicationGateways()
                .startAsync(this.resourceGroupName(), this.name())
                .flatMap(new Func1<Void, Observable<ApplicationGateway>>() {
                    @Override
                    public Observable<ApplicationGateway> call(Void aVoid) {
                        return refreshAsync();
                    }
                }).toCompletable();
}
```

This way both sync and async will have same behavior.

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java [hunk 2]
```diff
@@ -1235,4 +1237,28 @@ public ApplicationGatewayTier tier() {
...
+    }
+
+    @Override
+    public Completable stopAsync() {
+        return this.manager().inner().applicationGateways().stopAsync(this.resourceGroupName(),  this.name()).toCompletable();
+    }
```
Comment (anuchandy) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java]: same comment as `startAsync` and `start`

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java (hunk 1 above)
Comment (martinsawicki) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java]: done

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java (hunk 2 above)
Comment (martinsawicki) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/ApplicationGatewayImpl.java]: done

**Pull request merged**